server/src/queues/createDatabase.ts

```diff
@@ -0,0 +1,105 @@
+import { dbTypeToDokkuPlugin } from './../graphql/utils';
+import { Worker, Queue } from 'bullmq';
+import createDebug from 'debug';
+import { pubsub } from './../index';
+import { config } from '../config';
+import { sshConnect } from '../lib/ssh';
+import { prisma } from '../prisma';
+
+const queueName = 'create-database';
+const debug = createDebug(`queue:${queueName}`);
+
+interface QueueArgs {
+  databaseName: string;
+  databaseType: string;
+  userId: string;
+}
+
+export const createDatabaseQueue = new Queue<QueueArgs>(queueName, {
+  defaultJobOptions: {
+    // Max timeout 20 minutes
+    timeout: 1.2e6,
+  },
+  connection: config.redisClient,
+});
+
+/**
+ * - Create db
+ */
+const worker = new Worker(
+  queueName,
+  async (job) => {
+    const { databaseName, databaseType, userId } = job.data;
+
+    const dbType = dbTypeToDokkuPlugin(databaseType);
+
+    debug(
+      `starting createDatabase queue for ${dbType} database called ${databaseName}`
+    );
+
+    const ssh = await sshConnect();
+    const res = await ssh.execCommand(`${dbType}:create ${databaseName}`, {
+      onStdout: (chunk) => {
+        pubsub.publish('DATABASE_CREATED', {
+          createDatabaseLogs: {
+            message: chunk.toString(),
+            type: 'stdout',
+          },
+        });
+      },
+      onStderr: (chunk) => {
+        pubsub.publish('DATABASE_CREATED', {
+          createDatabaseLogs: {
+            message: chunk.toString(),
+            type: 'stderr',
+          },
+        });
+      },
+    });
+
+    const createdDb = await prisma.database.create({
+      data: {
+        name: databaseName,
+        type: databaseType,
+        user: {
+          connect: {
+            id: userId,
+          },
+        },
+      },
+    });
+
+    debug(
+      `finishing createDatabase queue for ${dbType} database called ${databaseName}`
+    );
+    if (!res.stderr) {
+      pubsub.publish('DATABASE_CREATED', {
+        createDatabaseLogs: {
+          message: `Successfully created DB with ID: ${createdDb.id}`,
```

Comment:
I think this would be cleaner if we only pass the id, the text can be done in the frontend only. That way you don't have to parse the message client side.

Also, what do you think about separating the end type to know if it was a failure or success via the type, eg: `end-success`, `end-failure` or `end:success`, `end:failure`?

Reply:
@pradel different types for end messages totally make sense!